gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +90,75 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_spliceai_grch38_ht() -> hl.Table:
+    """
+    Create a Hail Table with SpliceAI scores for GRCh38.
+
+    SpliceAI scores are from the following resources:
+        - Precomputed SNVs: spliceai_scores.masked.snv.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - Precomputed indels: spliceai_scores.masked.indel.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - gnomAD v4 indels: gnomad_v4_new_indels.spliceai_masked.vcf.bgz,
+          computed on v4 by Illumina.
+
+    :return: Hail Table with SpliceAI scores for GRCh38.
+    """
+    snvs_path = "gs://gnomad-insilico/spliceai/spliceai_scores.masked.snv.hg38.vcf.bgz"
+    indels_path = (
+        "gs://gnomad-insilico/spliceai/spliceai_scores.masked.indel.hg38.vcf.bgz"
+    )
+    new_indels_path = "gs://gnomad-insilico/spliceai/spliceai_scores.masked.gnomad_v4_new_indels.hg38.vcf.bgz"
+    header_file_path = "gs://gnomad-insilico/spliceai/spliceai.vcf.header"
+
+    def import_spliceai_vcf(path: str) -> hl.Table:
+        """
+        Import SpliceAI VCF into Hail Table.
+
+        :param str path: Path to SpliceAI VCF.
+        :return: Hail MatrixTable with SpliceAI scores.
+        """
+        ht = hl.import_vcf(
+            path,
+            force_bgz=True,
+            header_file=header_file_path,
+            reference_genome="GRCh38",
+            contig_recoding=NO_CHR_TO_CHR_CONTIG_RECODING,
+            skip_invalid_loci=True,
+            min_partitions=1000,
+        ).rows()
+        return ht
+
+    logger.info("Importing vcf of SpliceAI scores into HT...")
+
+    spliceai_snvs = import_spliceai_vcf(snvs_path)
+    spliceai_indels = import_spliceai_vcf(indels_path)
+    spliceai_new_indels = import_spliceai_vcf(new_indels_path)
+
+    ht = spliceai_snvs.union(spliceai_indels).union(spliceai_new_indels)
```

Comment:
So just for consistency, can you do what you did on L88? I'm not sure if I missed this before but for the union on L88, does hail automatically know the second arg is a table and not union's unify arg?

Reply:
Yes, `union` is to append rows of multiple HT, I didn't do that because it was `union_rows` for MT.